wee_slack.py

```diff
@@ -1277,12 +1292,17 @@ def edit_nth_previous_message(self, n, old, new, flags):
                 self.eventrouter.receive(s)
 
     def my_last_message(self, msgno):
-        for key in self.main_message_keys_reversed():
-            m = self.messages[key]
+        if type(msgno) is not int:
+            m = self.hashed_messages[msgno[1:]]
             if "user" in m.message_json and "text" in m.message_json and m.message_json["user"] == self.team.myidentifier:
-                msgno -= 1
-                if msgno == 0:
-                    return m.message_json
+                return m.message_json
+        else:
+            for key in self.main_message_keys_reversed():
+                m = self.messages[key]
+                if "user" in m.message_json and "text" in m.message_json and m.message_json["user"] == self.team.myidentifier:
```

Comment:
You can use `.get` instead of checking if `user` is present first. And I don't think we need to check text? (I know you just used the same check as is already here). So:

```suggestion
                if m.message_json.get("user") == self.team.myidentifier:
```

Reply:
Changed on both occurences